src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -137,6 +137,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DocumentationFile />
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(PublishDocumentationFile)' == ''">
+    <PublishDocumentationFile Condition="'$(GenerateDocumentationFile)' == 'true'">true</PublishDocumentationFile>
```

Comment:
>  if you haven an app and library as ProjectReference, the documentation file of the library is published with the app

Can you add a test for this?  I think using the `KitchenSink` project would be the easiest.  You could probably change your two tests below to use that TestAsset, and be able to test all these scenarios pretty easy.